### PR TITLE
Fixes #3026 - Handling links to openstreetmap.

### LINKED ISSUE #3026: Show map along with the link for map type response
**Actual Behaviour**
- Currently only the link is being shown for the map.

**Expected Behaviour**
- Show map along with the link as was shown previously.

**Steps to reproduce it**
Type `Where is Singapore?` into the chat box.

**Screenshots of the issue**
Screenshot 2019-11-07 at 12 00 03 PM

**Would you like to work on the issue?**
Anyone is free to pick this task.

### PR DESCRIPTION
**Pull request by AuraOfDivinity**
Fixes #3026 

Changes: 
Currently when the frontend sends a request containing a question with a map, the returned response contains an action type of an 'anchor' whereas the expected action type is map. 

Because of this, the returned response is handled under the 'anchor' case in the generateMessageBubbleUtils.js

An if condition was added inside the 'anchor' case to check for actions with a link referring to openstreetmap.org . If so the latitude and longitude were extracted from the link and handled in the same manner a map action type is handled inside the 'map' case.

**This is by no manes a permanent fix and must be removed once the issues related to the susi backend are fixed.**

Demo Link: https://pr-3027-fossasia-susi-web-chat.surge.sh

Screenshots of the change:

FixedMaps

### CODE REVIEW

File: src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js [hunk 1]
```diff
@@ -377,21 +376,75 @@ export const generateMessageBubble = (
...
-            link,
-            message,
-            latestUserMsgID,
-            showFeedback,
-          ),
-        );
+      case 'anchor':
+        {
+          const { link, text } = action;
+          if (link.substring(0, 25) === 'https://www.openstreetmap') {
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js]: Check the case where `map` was previously implemented and remove it from there. Also, write this logic into a function and just call that function from here.

File: src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js [hunk 2]
```diff
@@ -366,34 +374,88 @@ export const generateMessageBubble = (
...
           ),
         );
+        // }
         break;
       }
+      case 'anchor':
+        {
+          const { link, text } = action;
+          if (checkMapAction(link)) {
+            let mapAnchor = null;
```
Comment (akshatnitd) [inline src/components/ChatApp/MessageListItem/generateMessageBubbleUtil.js]: Move this entire logic of getting the lat long and other calculation to the other function defined above.

**Comment by AuraOfDivinity** [comment]
@akshatnitd @samagragupta  Made the requested changes. Kindly have a look and let me know if anything else needs to be done.

**Comment by AuraOfDivinity** [comment]
@akshatnitd Closing because of https://github.com/fossasia/susi.ai/pull/3080 . Kindly take a look at the new PR and let me know if new changes are required.

**Comment by AuraOfDivinity** [comment]
@akshatnitd Closing because of https://github.com/fossasia/susi.ai/pull/3080 . Kindly take a look at the new PR and let me know if new changes are required.

**Pull request closed without merging**